Source/CarthageKit/GitHub.swift

```diff
@@ -55,3 +69,235 @@ extension GitHubRepository: Printable {
 		return "\(owner)/\(name)"
 	}
 }
+
+/// Represents a Release on a GitHub repository.
+public struct GitHubRelease: Equatable {
+	/// The unique ID for this release.
+	public let ID: Int
+
+	/// The name of this release.
+	public let name: String
+
+	/// The name of the tag upon which this release is based.
+	public let tag: String
+
+	/// Whether this release is a draft (only visible to the authenticted user).
+	public let draft: Bool
+
+	/// Whether this release represents a prerelease version.
+	public let prerelease: Bool
+
+	/// Any assets attached to the release.
+	public let assets: [Asset]
+
+	/// An asset attached to a GitHub Release.
+	public struct Asset: Equatable, Hashable, Printable, JSONDecodable {
+		/// The unique ID for this release asset.
+		public let ID: Int
+
+		/// The filename of this asset.
+		public let name: String
+
+		/// The MIME type of this asset.
+		public let contentType: String
+
+		/// The URL at which the asset can be downloaded directly.
+		public let downloadURL: NSURL
+
+		public var hashValue: Int {
+			return ID.hashValue
+		}
+
+		public var description: String {
+			return "Asset { name = \(name), contentType = \(contentType), downloadURL = \(downloadURL) }"
+		}
+
+		public static func create(ID: Int)(name: String)(contentType: String)(downloadURL: NSURL) -> Asset {
+			return self(ID: ID, name: name, contentType: contentType, downloadURL: downloadURL)
+		}
+
+		public static func decode(j: JSONValue) -> Asset? {
+			return self.create
+				<^> j <| "id"
+				<*> j <| "name"
+				<*> j <| "content_type"
+				<*> j <| "browser_download_url"
+		}
```

Comment:
What do you think of [`check`](https://gist.github.com/andymatuschak/2b311461caf740f5726f#file-gistfile1-swift-L13-L16) as a more readable (and not very abstract) abstraction for handling this sort of thing?